### PR TITLE
show documentSymbols in a StructureView

### LINKED ISSUE #128: Support of `textDocument/documentSymbol`?
Any plans or ideas for $subject?

### PR DESCRIPTION
**Pull request by swissiety**
uses LSP to populate IntelliJs StructureView
- [ ] link to referenced Location on click

Example
intellij_documentHighlight

closes #128

### CODE REVIEW

File: src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java [hunk 1]
```diff
@@ -31,6 +29,7 @@
 import org.wso2.lsp4intellij.utils.FileUtils;
 
 import javax.swing.*;
+import javax.tools.Diagnostic;
```
Comment (nixel2007) [inline src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java:32]: Is this a correct import?

**Comment by Trias** [comment]
>  link to referenced Location on click
@swissiety maybe you can use this commit https://github.com/ballerina-platform/lsp4intellij/pull/232/commits/f77a9eae70e99aec09f88d2067a8be9c44f9b2c2 for goto location (not merged yet)

**Comment by swissiety** [comment]
The StructureView *should* link automatically via the given PSIElement - but it doesn't.

**Comment by NipunaRanasinghe** [comment]
@swissiety Is this still WIP or ready to review?

**Comment by swissiety** [comment]
The structureview presents the symbols but I still havent figured why its not navigating to the referenced symbol position - I'll try to find some time in march to work on that

**Pull request closed without merging**